coalib/bearlib/languages/definitions/TinyBasic.py

```diff
@@ -0,0 +1,18 @@
+from coalib.bearlib.languages.Language import Language
+
+
+@Language
+class TinyBasic:
+    aliases = 'tb',
+    extensions = '.tb',
+    comment_delimiters = 'REM', "'"
+    string_delimiters = {'"': '"', "'": "'"}
+    encapsulators = {'(': ')'}
+    keywords = [
```

Comment:
I think there are other keywords. You can refer here http://tinybasic.de/download/TinyBasicV240.pdf

Reply:
I think that may be a heavily modified version of TinyBasic designed specifically for programming micro controllers. TinyBasic seems to have been written in the 1970s (see [this guide](http://users.telenet.be/kim1-6502/tinybasic/tbum.html) from 1976), and the guide that you linked to is from 2009. [Wikipedia](https://en.wikipedia.org/wiki/Tiny_BASIC) seems to corroborate the 1970s timeline, and it doesn't refer to any recently updated version of TinyBasic.

In addition to that, the creator of TinyBasic according to Wikipedia is Dennis Allison (from the US), while the creator of the TinyBasic [from the guide that you linked](http://www.tinybasic.de/contact.htm) is Dr.-Ing. (from Germany). The sample code published on [that guide's site](http://www.tinybasic.de/samples.htm) also does not seem to match in style to samples provided in [this original guide](http://users.telenet.be/kim1-6502/tinybasic/tbum.html).

Reply:
Ahh that might be the case. What do you say @pareksha . Should we include this ?

Reply:
Tiny Basic is a language that is specifically designed for micro controllers only (that is what I read, correct me if I am wrong).

I see that in the link with Dr. -Ing that the version of language is 2.4.

Read about the new keywords and changes in the language. If there's a major change with the incoming of new versions, include those changes, but with a comment that this change is applicable on and after <version name> version.

Reply:
Sounds good, I will take care of those changes today.

Reply:
Changes have been added!

Reply:
the TinyBasic in that pdf is also umaintained. This is a 2009 snapshot
https://web.archive.org/web/20090410204355/http://www.tinybasic.de/downloads.htm

So I thikn it is not worth considering as part of the TinyBasic history .